src/CFamily.UnitTests/CMake/VsDevCmdEnvironmentProviderTests.cs

```diff
@@ -0,0 +1,357 @@
+            };
+
+            var context = new ProcessContext(simulatedWorkCallback: writeOutputOp);
+
+            // Act
+            var actual = context.TestSubject.Get("any");
+
+            actual.Should().NotBeNull();
+            actual.Count.Should().Be(2);
+            actual.Keys.Should().BeEquivalentTo("key1", "key2");
+            actual["key1"].Should().Be("value1");
+            actual["key2"].Should().Be("value with spaces");
+        }
+
+        [TestMethod]
+        public void Get_DataProcessing_TimeoutOccurs_NullReturned()
+        {
+            ProcessContext.SimulateWorkInProcess writeOutputOp = (context) =>
+            {
+                context.WriteBeginCapture();
+                context.WriteToOutput("key1=value1");
+                context.WriteToOutput("key2=value2");
+
+                return false; // hasExited = false i.e. timeout
+            };
+
+            var context = new ProcessContext(simulatedWorkCallback: writeOutputOp);
+
+            // Act
+            var actual = context.TestSubject.Get("any");
+
+            actual.Should().BeNull();
+            context.Logger.AssertPartialOutputStringExists(Resources.VsDevCmd_TimedOut);
+        }
+
+        [TestMethod]
+        public void Get_DataProcessing_NoSettingsCaptured_NullReturned()
+        {
+            ProcessContext.SimulateWorkInProcess writeOutputOp = (context) =>
+            {
+                context.WriteBeginCapture();
+                context.WriteToOutput("not a valid setting");
+                context.WriteEndCapture();
+                return true; // hasExited = true i.e. completed successfully
+            };
+
+            var context = new ProcessContext(simulatedWorkCallback: writeOutputOp);
+
+            // Act
+            var actual = context.TestSubject.Get("any");
+
+            actual.Should().BeNull();
+            context.Logger.AssertPartialOutputStringExists(Resources.VsDevCmd_NoSettingsFound);
+        }
+
+        [TestMethod]
+        public void Get_NonCriticalException_IsSuppressed()
+        {
+            var context = new ProcessContext();
+
+            context.FileSystem.Reset();
+            context.FileSystem.Setup(x => x.File.Exists(It.IsAny<string>()))
+                .Throws(new InvalidCastException("thrown from test"));
+
+            var actual = context.TestSubject.Get("any");
+
+            actual.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Get_CriticalException_IsNotSuppressed()
+        {
+            var context = new ProcessContext();
+
+            context.FileSystem.Reset();
+            context.FileSystem.Setup(x => x.File.Exists(It.IsAny<string>()))
+                .Throws(new StackOverflowException("thrown from test"));
+
+            Action act = () => context.TestSubject.Get("any");
+
+            act.Should().ThrowExactly<StackOverflowException>().And.Message.Should().Be("thrown from test");
+        }
+
+        private class ProcessContext
```

Comment:
The tests are relatively straightforward, because the setup complexity is all in this class.

The class falls into two main parts:
* a set of Mocks that are always created for the process factory etc.
* a more complicated "simulate process execution" part that is only configured if the test provides a callback to simulate the process doing something.